src/main/java/com/johan/blignaut/entelect/vitalityincubator/service/GeoDBMappingServiceImpl.java

```diff
@@ -15,11 +15,13 @@
 
 @Service
 public class GeoDBMappingServiceImpl implements GeoDBMappingService {
+
     @Override
-    public GeoDBServer mapClientToServerGeoDB(GeoDBClient geoDBClient) {
+    public GeoDBServer mapClientToServerGeoDB(GeoDBClient geoDBClient, String filterByName) {
         GeoDBServer geoDBServer = new GeoDBServer();
         List<CityServer> citiesList = geoDBClient.getData()
                 .stream()
```

Comment:
parallel stream here.

It takes advantage of multi-core processors to perform operations in parallel, potentially improving the overall performance for computationally intensive tasks or when dealing with large data sets.